shopelectro/views/catalog.py

```diff
@@ -25,6 +26,62 @@ def get_products_count(request):
     return PRODUCTS_ON_PAGE_MOB if mobile_view else PRODUCTS_ON_PAGE_PC
 
 
+def get_view_type(request):
+    view_type = request.session.get('view_type', 'tile')
+    assert view_type in ['list', 'tile']
+    return view_type
+
+
+def get_catalog_context(request, category, raw_tags, page_number, per_page, sorting_index):
+    all_tags = newcontext.Tags(models.Tag.objects.all())
+
+    if raw_tags:
+        selected_tags = newcontext.tags.Checked404Tags(
+            newcontext.tags.ParsedTags(
+                tags=all_tags,
+                raw_tags=raw_tags,
+            )
+        )
+    else:
+        selected_tags = newcontext.Tags(models.Tag.objects.none())
+
+    products = newcontext.products.OrderedProducts(
+        products=newcontext.products.TaggedProducts(
```

Comment:
maybe we can drop `*Products` suffix here?
We are already at the `products` module.
And using it will be clear too: `newcontext.products.Tagged` for example

Reply:
I will create todo for it